### ISSUE TITLE
SetMargins doesn't apply correct margins

### ISSUE THREAD

**Issue by Llermy** [gree/unity-webview#918]
I seem unable to get around with the SetMargins method. To test it out, I've created a very simple WebViewObject which loads sample.html from the samples, to which I've set margins by:
```
webViewObject.SetMargins(Screen.width / 2, Screen.height / 2, 0, 0);
```

This should produce a WebView window on the bottom right corner of the viewport corresponding to a fourth of it (the red part of the following picture):

![屏幕截图 2023-04-27 185300](https://user-images.githubusercontent.com/12306235/234934454-ee049768-177d-4f13-996e-185dd5b37f12.png)

However, SetMargins seems to be setting wrong margins for both the left and top components. Making measurements in the case of the example WebGL picture I've posted, half the width of the screen is 600 but the resulted left margin is 750 instead. Half the height of the screen is 375 but the resulting top margin seems to be 280. Is something that I should take into account apart from the guideline in the git respository?

The whole script is like this:
```
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class WebPanel : MonoBehaviour
{
    public string Url;

    private WebViewObject webViewObject;

    // Start is called before the first frame update
    IEnumerator Start()
    {
        webViewObject = (new GameObject("WebViewObject")).AddComponent<WebViewObject>();
        webViewObject.Init();

#if UNITY_WEBPLAYER || UNITY_WEBGL
        var js = @"
                    window.Unity = {
                        call:function(msg) {
                            parent.unityWebView.sendMessage('WebViewObject', msg);
                        }
                    };
                ";
#else
                var js = "";
#endif
        webViewObject.EvaluateJS(js + @"Unity.call('ua=' + navigator.userAgent)");

        var bounds = GetRectTransformBounds(GetComponent<RectTransform>());
        webViewObject.SetMargins(Screen.width / 2, Screen.height / 2, 0, 0);
        webViewObject.SetVisibility(true);
        webViewObject.LoadURL("StreamingAssets/" + Url.Replace(" ", "%20"));

        yield break;
    }
}
```

**Comment by KojiNakamaru** [comment]
I guess you use Unity 2020.1.0f1 or later. Do you specify `unity-webview-2020` as described in https://github.com/gree/unity-webview/tree/3647674386e3c844aa416fc744905b8286c56616#webgl ?

**Comment by Llermy** [comment]
I'm using 2019.3.14f1 and not using unity-webview-2020 therefore :)

**Comment by Llermy** [comment]
I think I've found the problem and solved it, at least in my case. The problem is that the real size of the canvas in HTML and Unity's render size may not be here, so the problem lies here:

```
var container = $('#unityContainer');
var width = container.width();
var height = container.height();
```

This retrieves the real size of the canvas in HTML, but the `left, top, right, bottom` parameters of the function are given with respect to Unity's render size, so the proportions computed afterwards are not right.

* Solution 1: just get Unity's render size width and height, which are set as attributes in the canvas element.
```
var container = $('[id="#canvas"]');
var width = container.attr("width");
var height = container.attr("height");
```
* Solution 2: multiply the container real width and height with the `window.devicePixelRatio`, as is done in the template unity-webview-2020, but independently of its check if there is a `unity-desktop` class, as there is none at least in Unity 2019.3.14f1.
```
var container = $('#unityContainer');
var width = container.width() * window.devicePixelRatio;
var height = container.height() * window.devicePixelRatio;
```

I am not sure if this can be applied to other Unity versions, devices and is general enough. But in my case, this solved the problem :)

**Comment by KojiNakamaru** [comment]
Thanks! I also investigated the issue and fixed it in #920, which is basically the same fix of yours (solution 2).

**Comment by Llermy** [comment]
Cool great job!
